### PR TITLE
Kdebonus

### PR DESCRIPTION
**Pull request by gcperk**
Hi @ateucher - this is the updated KDE bonus content. Let me know if there is anything that needs tweaking. 
Thanks!

### CODE REVIEW

File: bonus-kde.qmd [hunk 1]
```diff
@@ -1,180 +1,231 @@
...
+  # convert this to a sf object 
+  c.mcp_sf <- st_as_sf(c.mcp)
+  
+  # plot with mapview
+  mapview::mapview(c.mcp_sf)
+  
+```
 
-#https://cran.r-project.org/web/packages/ks/vignettes/kde.pdf
+The advantage of using this package is that it provides many more outputs and enables us to customise the information we want. For example we can extract multiple homeranges based on varying levels of specificity; i.e. percent = 75%.
```
Comment (ateucher) [inline bonus-kde.qmd]:
```suggestion
The advantage of using the adhabitatHR package is that it provides many more outputs and enables us to customise the information we want. For example we can extract multiple homeranges based on varying levels of specificity; i.e. percent = 75%.
```

File: bonus-kde.qmd [hunk 2]
```diff
@@ -1,180 +1,231 @@
...
 
-Hpi1 <- Hpi(x=x)
-Hpi2 <- Hpi.diag(x = x)
+Other parameters include:
 
+-   Kernel Type: The type of kernel is limited to Gaussian (bivariate normal), quadratic or normal. We used the bivariate normal model as default.
 
-fhat.pi1 <-kde(x=x,H=Hpi1) 
-fhat.pi2 <-kde(x=x,H=Hpi2)
+-   Grid/extent: These determines the area or extent over which the home range will be estimated. This is a mix of fine scale and time consuming processing and faster blocky resolution over a continuous surface. As a rule of thumb you GME suggests: take the square root of the x or y variance value (whichever is smaller) and divide by 5 or 10 (I usually round to the nearest big number - so 36.7 becomes 40). Before using this rule of thumb value calculate how many cells this will result in for the output (take the width and height of you input points, divide by the cell size, and multiply the resulting numbers together). If you get a value somewhere between 1-20 million, then you have a reasonable value.
```
Comment (ateucher) [inline bonus-kde.qmd]: What is GME?

File: bonus-kde.qmd [hunk 3]
```diff
@@ -1,180 +1,231 @@
...
 
-#https://bookdown.org/egarpor/NP-UC3M/kde-i-kde.html
+```{r}
+#  Create SpatialPointsDataFrame object
+  tdfid <- tdf |>  dplyr::select(location.long, location.lat, animal.id)
+
+  coordinates(tdfid) <- c("location.long", "location.lat")
+  proj4string(tdfid) <- CRS( "+proj=longlat +datum=WGS84 +units=m +no_defs" )
+  
+  tdfgeo <- spTransform(tdfid, CRS("+init=epsg:3005")) # transform
```
Comment (ateucher) [inline bonus-kde.qmd]: Can we just do `as(bou, "Spatial")`?

```suggestion
  tdfgeo <- bou |>  
    dplyr::select(animal.id) |>
    as("Spatial")
```

File: bonus-kde.qmd [hunk 4]
```diff
@@ -1,180 +1,231 @@
...
 
+#### 3.1 kde: h reference parameter
 
+```{r}
+  tdfu <- tdf  |>  dplyr::select(location.long, location.lat, season)
+    
+  # Create a SpatialPointsDataFrame by defining the coordinates
+  coordinates(tdfu) <- c("location.long", "location.lat")
+  proj4string(tdfu) <- CRS( "+proj=longlat +datum=WGS84 +units=m +no_defs" )
+  tdfgeo <- spTransform(tdfu, CRS("+init=epsg:3005")) # Transform to UTM
```
Comment (ateucher) [inline bonus-kde.qmd]: As suggested above, I think this works?

```suggestion
tdfgeo <- bou |>  
  dplyr::select(animal.id) |>
  as("Spatial")
```

File: bonus-kde.qmd [hunk 5]
```diff
@@ -1,180 +1,231 @@
...
+  ver50_sf<- st_as_sf(ver50)
+ 
+  # plot the outputs 
+  mapview(ver50_sf, zcol = "id") 
+  mapview (ver75_sf, zcol = "id")
+  mapview (ver95_sf, zcol = "id")
+  
+  plot(st_geometry(ver95.2.sf),col = "yellow") 
+  plot(st_geometry(ver75.2.sf),col = "blue", add = TRUE)
+  plot(st_geometry(ver50.2.sf),col = "purple", add = TRUE)
```
Comment (ateucher) [inline bonus-kde.qmd]: Objects names `.2.sf` don't exist, I think you mean these?

```suggestion
  plot(st_geometry(ver95_sf),col = "yellow") 
  plot(st_geometry(ver75_sf),col = "blue", add = TRUE)
  plot(st_geometry(ver50_sf),col = "purple", add = TRUE)
```

File: bonus-kde.qmd [hunk 6]
```diff
@@ -1,180 +1,231 @@
...
-
-
-
-fhat.pi11<-kde(x=x,H=Hscv1) 
-fhat.pi22<-kde(x=x,H=Hscv2)
-
-plot(fhat.pi11)
-plot(fhat.pi22)
+```{r}
+  kde_lscv  <- kernelUD(tdfgeo, h = "LSCV", kern = c("bivnorm"), grid = 500, extent = 2)
```
Comment (ateucher) [inline bonus-kde.qmd]: will we plot/look at this one?

File: bonus-kde.qmd [hunk 7]
```diff
@@ -1,180 +1,231 @@
...
+  geom_sf(
+    aes(colour = id), 
+    alpha = 0.1
+  ) + 
+  scale_colour_viridis_d() + 
+  facet_wrap(vars(h)) +
+  theme_bw()
+  
+  
+```
```
Comment (ateucher) [inline bonus-kde.qmd:285]: Will you end with recommendations for what to use under what circustances?

**Review by ateucher — APPROVED** [review]
This is super cool, very nice. A few code suggestions and comments

File: bonus-kde.qmd (hunk 2 above)
Comment (gcperk) [reply bonus-kde.qmd]: fixed

File: bonus-kde.qmd (hunk 4 above)
Comment (gcperk) [reply bonus-kde.qmd]: Yes I just tested and works great!

File: bonus-kde.qmd (hunk 5 above)
Comment (gcperk) [reply bonus-kde.qmd]: Thanks for catching that

**Pull request merged**